pymeasure/instruments/common_base.py

```diff
@@ -0,0 +1,569 @@
+        with special name and copy them at instance level
+        Internal method, not intended to be accessed at user level."""
+        special_names = []
+        dynamic_params = tuple(set(self._fget_params_list + self._fset_params_list))
+        # Check whether class variables of DynamicProperty type are present
+        for obj in (self,) + self.__class__.__mro__:
+            for attr_name, attr in obj.__dict__.items():
+                if isinstance(attr, DynamicProperty):
+                    special_names += [attr_name + "_" + key for key in dynamic_params]
+        # Check if special variables are defined at class level
+        for obj in (self,) + self.__class__.__mro__:
+            for attr in obj.__dict__:
+                if attr in special_names:
+                    # Copy class special variable at instance level, prefixing reserved_prefix
+                    setattr(self, self.__reserved_prefix + attr, obj.__dict__[attr])
+        return special_names
+
+    def __setattr__(self, name, value):
+        """ Add reserved_prefix in front of special variables."""
+        if hasattr(self, '_special_names'):
+            if name in self._special_names:
+                name = self.__reserved_prefix + name
+        super().__setattr__(name, value)
+
+    def __getattribute__(self, name):
+        """ Prevent read access to variables with special names used to
+        support dynamic property behaviour."""
+        if name in ('_special_names', '__dict__'):
+            return super().__getattribute__(name)
+        if hasattr(self, '_special_names'):
+            if name in self._special_names:
+                raise AttributeError(
+                    f"{name} is a reserved variable name and it cannot be read")
+        return super().__getattribute__(name)
+
+    # Channel management
+    def add_child(self, cls, id, collection="channels", prefix="ch", **kwargs):
+        """Add a child to this instance and return its index in the children list.
+
+        The newly created child may be accessed either by the index in the
+        children list or by the created attribute.
+        The fifth channel of `instrument` with id "F" has two access options:
+        :code:`instrument.channels[4] == instrument.ch_F`
+
+        .. note::
+
+            Do not change the default `collection` or `prefix` parameter, unless
+            you have to distinguish several collections of different children,
+            e.g. different channel types (analog and digital).
+
+        :param cls: Class of the channel.
+        :param id: Child id how it is used in communication, e.g. `"A"`.
+            If `None`, the child will be added directly under the collection name.
+        :param collection: Name of the collection of children, used for the list.
+        :param prefix: Collection prefix for the attributes, e.g. `"ch"`
+            creates attribute `self.ch_A`. An underscore separates prefix from id.
```

Comment:
We should put the underscore into the default prefix (i.e. `ch_`) and then just concatenate. Same effect in the default case, more flexible for the user.

Reply:
Done